src/PhpDoc/PhpDocNodeResolver.php

```diff
@@ -185,12 +185,24 @@ public function resolveMethodTags(PhpDocNode $phpDocNode, NameScope $nameScope):
 					);
 				}
 
+				$templates = [];
```

Comment:
I figured out what needs to be done.

We need to look at `$tagValue->templateTypes` first and when it's non empty, we need to call `$nameScope->withTemplateTypeMap(...)`. And pass that object instead of currently passing `$nameScope` basically to all `$this->typeNodeResolver->resolve(...)` inside this foreach in `resolveMethodTags`.